tests/testthat/test-interact.R

```diff
@@ -127,3 +127,38 @@ test_that("run_safely_loudly works for result", {
 
    log_remove()
 })
+
+test_that("run_file uses a child of the global environment for execution", {
+
+   expect_equal(capture.output(globalenv()),
+                capture.output(logrx:::run_file(test_path("ref", "run_file_test_parent_env.R"))))
+
+})
+
+test_that("run_file makes no changes are made to the global environment or options during execution", {
```

Comment:
So there's a fallacy in the logic of this test because of the way environmental bindings work. Much like mutability in Python:

```r
> env1 <- new.env()
> env2 <- env1
> 
> assign("test", TRUE, envir=env1)
> 
> print(env1$test)
[1] TRUE
> print(env2$test)
[1] TRUE
```

I think it would be sufficient here to do this instead:

```r
   pre_global <- ls(envir=globalenv())
   pre_ops <- options()

   logrx:::run_file(test_path("ref", "run_file_test.R"))

   post_global <- ls(envir=globalenv())
   post_ops <- options()

   expect_identical(pre_global, post_global)
   expect_identical(pre_ops, post_ops)
```

Reply:
I meant to use `expect_identical()`, will update now

Reply:
so expect_identical isn't hte issue, it's that the environments in the current code will _always_ be identical, because when you do pre_global <- globalenv(), if globalenv() is updated, pre_global will also be updated. It's a link, not a copy.